Kudu.Services/Deployment/PushDeploymentController.cs

```diff
@@ -869,6 +881,39 @@ private string GetZipDeployPathInfo()
             return pathmsg;
         }
 
+        private void BackupErrorZip(ArtifactDeploymentInfo zipDeploymentInfo)
+        {
+            try
+            {
+                // File created in D:\home\data\SitePackages for current deployment
+                var filePath = Path.Combine(_environment.SitePackagesPath, zipDeploymentInfo.ArtifactFileName);
+
+                // Store only if zip file size < 100 MB
+                long fileBytes = new FileInfo(filePath).Length;
+                float fileMB = (float)fileBytes / (1_024 * 1_024);
+                if (fileMB < 100)
+                {
+                    // Make sure D:\home\LogFiles\kudu\deployment\BackupErrorZip exists
+                    var errorDirPath = Path.Combine(_environment.DeploymentTracePath, "BackupErrorZip");
+                    FileSystemHelpers.EnsureDirectory(errorDirPath);
+
+                    // Store only 1 zip file with error
+                    if (FileSystemHelpers.GetFiles(errorDirPath, "*.zip").Length == 0)
+                    {
+                        var errorFilePath = Path.Combine(errorDirPath, zipDeploymentInfo.ArtifactFileName);
+                        using (_tracer.Step("{0}: Saving zip file with error to {1} of size {2} MB.", nameof(BackupErrorZip), errorFilePath, fileMB))
+                        {
+                            File.Copy(filePath, errorFilePath);
```

Comment:
Probably use `FileSystemHelpers.CopyFile` instead of `File.Copy` for better testability.